### PR TITLE
Adds bitbucket_server_trigger_config field to cloudbuild trigger

### PR DESCRIPTION
**Pull request by mariomachado94**
resource in order to support bitbucket server triggers.

Resolves https://github.com/hashicorp/terraform-provider-google/issues/11499

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
cloudbuild: added `bitbucket_server_trigger_config` field to `google_cloudbuild_trigger` resource
```

### CODE REVIEW

File: mmv1/products/cloudbuild/terraform.yaml [hunk 1]
```diff
@@ -65,6 +65,10 @@ overrides: !ruby/object:Overrides::ResourceOverrides
           installation_id: 31300675
           pat_secret: '"projects/gcb-terraform-creds/secrets/github-pat/versions/latest"'
           repo_uri: '"https://github.com/gcb-repos-robot/tf-demo.git"'
+      - !ruby/object:Provider::Terraform::Examples
+        name: "cloudbuild_trigger_bitbucket_server"
+        primary_resource_id: "bbs-trigger"
+        skip_test: true
```
Comment (ScottSuarez) [inline mmv1/products/cloudbuild/terraform.yaml:71]: Is there a reason we are skipping test here? This prevents it from running in our CI unfortunately

File: mmv1/templates/terraform/examples/cloudbuild_trigger_bitbucket_server.tf.erb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  location    = "us-central1"
+
+  bitbucket_server_trigger_config {
+    repo_slug = "terraform-provider-google"
+    project_key = "STAG"
+    bitbucket_server_config_resource = "projects/123456789/locations/us-central1/bitbucketServerConfigs/myBitbucketConfig"
+    push {
+        branch = "^master$"
+    }
+  }
```
Comment (ScottSuarez) [inline mmv1/templates/terraform/examples/cloudbuild_trigger_bitbucket_server.tf.erb:12]: Looks like we only flex part of the changes... Could we another test and flex as many fields as possible?

**Review by ScottSuarez — CHANGES_REQUESTED** [review]
You'll need to pull latest changes from main one you make your changes.. We've modified our CI recently and it requires it.

File: mmv1/templates/terraform/examples/cloudbuild_trigger_bitbucket_server.tf.erb (hunk 2 above)
Comment (mariomachado94) [reply mmv1/templates/terraform/examples/cloudbuild_trigger_bitbucket_server.tf.erb]: Yes, updated tests.

File: mmv1/products/cloudbuild/terraform.yaml (hunk 1 above)
Comment (mariomachado94) [reply mmv1/products/cloudbuild/terraform.yaml]: Configured 2 reals tests that pass. Done. I thought our API would return an error on creation unless we had some valid bitbucket server resources created. But after checking I realized we can create a trigger with non-existent resource references.

**Review by ScottSuarez — APPROVED** [review]
Thanks for the work Mario :)

**Pull request merged**